internal/pkg/checks/hostRulesCheck.go

```diff
@@ -22,29 +22,116 @@ import (
 )
 
 func init() {
+	hostRulesCheck.AddCheck(hostRulesExactMatchCheck)
```

Comment:
It seems like we are building some small form of generic test framework here. (for example, there is also filtering etc).

- We should have a bigger conversation about whether or not it makes sense as this could easily replicate things that existing frameworks have, such as parallel execution, result collection etc, that are implemented by ginkgo or the go test framework.
- It would be good to split the test description from the framework itself.

Reply:
Differing this change to the "framework" issue.